### PR TITLE
an implementation for #306 and that fixes #319 as well

### LINKED ISSUE #319: The toolbar button is very wide
The _Enter/Exit Live Reveal Slideshow_ button in the toolbar is very wide.

screen shot 2017-11-19 at 21 53 18

Being a button, it already implies a command. Further, the button is never actually used for exiting, only entering the slideshow.

This is a cosmetic (ie. important) issue, and becomes more of a functional issue when the user is installing other extensions which add buttons to the toolbar.

### PR DESCRIPTION
**Pull request by parmentelat**
this is mostly a big cleanup that uses jupyter's actions mechanism for everything including the toolbar button

once I was done I realized it also fixed #319, in a radically different way, I guess it will obsolete #320, sorry about that, it was not the intention

### CODE REVIEW

File: rise/static/main.js [hunk 1]
```diff
@@ -806,62 +818,56 @@ define([
...
-        help: 'edit all cells (all cells go to edit mode)',
-        handler: function() {
-          Jupyter.notebook.get_cells().forEach(function(cell){
-	    cell.unrender();
-          })
-        }
-      },
-      "edit-all-cells", "RISE");
+    // newtype can be any of 'slide' 'subslide' 'fragment' 'note' 'skip'
+    function toggle_slide_type(newtype) {
```
Comment (damianavila) [inline rise/static/main.js]: this should be `new_type` OR change those references below to `newtype`, otherwise you will get an error...

**Comment by damianavila** [comment]
> this is mostly a big cleanup that uses jupyter's actions mechanism for everything including the toolbar button

Great!

> once I was done I realized it also fixed #319, in a radically different way, I guess it will obsolete #320, sorry about that, it was not the intention

I know, and I think your solution is better so I decided to close the other PR giving the author all my gratitude for their contribution (thanks again @xmacex).

File: rise/static/main.js (hunk 1 above)
Comment (parmentelat) [reply rise/static/main.js]: good catch, thanks for spotting that one - fixed in 72c8586

**Comment by parmentelat** [comment]
apologies and thanks to @xmacex from my end as well :)

**Comment by parmentelat** [comment]
Hi Damian

I think I have fixed the issue that you pointed out and so as far as I am concerned this could be merged at your best convenience

let me know if I'm missing something


-- Thierry

**Comment by damianavila** [comment]
> let me know if I'm missing something

Just one more thing and we will merge it: https://github.com/damianavila/RISE/blob/master/doc/customize.rst

Please update that section with the changes we introduced here (pretty sure there is some changes needed around the default value we used to have).

**Comment by parmentelat** [comment]
Hi Damian

I have reviewed `customize.rst` as per your request, and made one last change in `main.js` so that now `smart-exec` is also registered as a jupyter action, which plays better with jupyter's helper tools

let me know if you request anything else before merging this

**Comment by damianavila** [comment]
Changes LGTM, let merge it and iterate if needed. Thanks again for you contributions!

**Pull request merged**